.github/workflows/teamcity-nightly-workflow.yaml

```diff
@@ -0,0 +1,102 @@
+name: "TeamCity: Create empty branch off tip of main to aid nightly-testing"
+
+
+# To ensure nightly tests/builds run on the same commit, we checkout and create a new branch from main for TeamCity to run builds on
+
+on:
+    workflow_dispatch:
+      inputs:
+        dayThreshold:
+          default: '3'
+    schedule:
+        - cron: '0 4 * * *'
+
+jobs:
+  create-branch:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@e69ef5462fd455e02edcaf4dd7708eda96b9eda0 # v7.0.0
```

Comment:
It'd be great to add an easy to understand `name` for this step, so it's obvious that it's creating a new branch for nightly testing